### PR TITLE
Start TinyCrypt deprecation

### PR DESCRIPTION
**Pull request by valeriosetti**
## Description

This PR:

- moves TinyCrypt library to the deprecation phase (as planned in #43712)
- updates the documentation (release-notes and migration-guide)

## Work in progress

Remove all usages of TinyCrypt from the Zephyr's codebase as required from the [API Lifecycle documentation](https://docs.zephyrproject.org/latest/develop/api/api_lifecycle.html#deprecated)

- https://github.com/zephyrproject-rtos/zephyr/pull/79653
- https://github.com/zephyrproject-rtos/zephyr/pull/79931

## To do

- add a new line in https://github.com/zephyrproject-rtos/zephyr/issues/77070

### CODE REVIEW

**Comment by rettichschnidi** [comment]
Maybe also mention that the project is no longer maintained upstream?

**Comment by ceolin** [comment]
Wonderful !

**Review by dkalowsk — DISMISSED** [review]

File: doc/releases/release-notes-4.0.rst [hunk 1]
```diff
@@ -56,6 +56,17 @@ Deprecated in this release
 
 * The :ref:`kscan_api` subsystem has been marked as deprecated.
 
+* Starting from this release the library is being marked as deprecated (:github:`79566`).
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:59]: `the library`? Maybe say which? :slightly_smiling_face:

File: doc/releases/release-notes-4.0.rst [hunk 2]
```diff
@@ -56,6 +56,17 @@ Deprecated in this release
...
+  The reasons for this are that (:github:`43712``):
+
+  * the upstream version of this library is unmaintained
+
+  * reduce the number of crypto libraries available in Zephyr (currently there are
+    3 different implementations: TinyCrypt, MbedTLS and PSA Crypto APIs).
+
+  PSA Crypto APIs are proposed as the de-facto standard for the future to perform
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:67]:
```suggestion
  The PSA Crypto API is now the de-facto standard to perform
```

File: doc/releases/release-notes-4.0.rst [hunk 3]
```diff
@@ -56,6 +56,17 @@ Deprecated in this release
 
 * The :ref:`kscan_api` subsystem has been marked as deprecated.
 
+* Starting from this release the library is being marked as deprecated (:github:`79566`).
+  The reasons for this are that (:github:`43712``):
+
+  * the upstream version of this library is unmaintained
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:62]:
```suggestion
  * the upstream version of this library is unmaintained.
```

File: doc/releases/release-notes-4.0.rst [hunk 4]
```diff
@@ -56,6 +56,17 @@ Deprecated in this release
 
 * The :ref:`kscan_api` subsystem has been marked as deprecated.
 
+* Starting from this release the library is being marked as deprecated (:github:`79566`).
+  The reasons for this are that (:github:`43712``):
```
Comment (tomi-font) [inline doc/releases/release-notes-4.0.rst:60]:
```suggestion
  The reasons for this are (:github:`43712``):
```

**Review by tomi-font — CHANGES_REQUESTED** [review]

File: doc/releases/release-notes-4.0.rst (hunk 1 above)
Comment (valeriosetti) [reply doc/releases/release-notes-4.0.rst:59]: Ops, I copied that from the `migration-guide` but in that document this bullet point is in a TinyCrypt specific section so the library is implicit

**Review by tomi-font — CHANGES_REQUESTED** [review]

**Review by tomi-font — APPROVED** [review]
Looks good. Just make sure to add the entry to the deprecation issue (could be done now?).

**Review by tomi-font — APPROVED** [review]

**Review by dkalowsk — APPROVED** [review]
I believe this one can be merged now, correct?  Or do you want to close this in favor of your Part1 PR?

**Comment by valeriosetti** [comment]
> I believe this one can be merged now, correct? Or do you want to close this in favor of your Part1 PR?

The idea is to have 3 separated PRs (this one, part1 and part2) to keep reviews simple. So if the current one is OK for all the reviewers we can merge it.

**Pull request merged**